### PR TITLE
Marketplace enhancement and new documentation!

### PR DESCRIPTION
**Pull request by Martins6**
I made some refactoring in the code deleting what was useless, fixed the documentation in the doc/ section, and also made some new enhancements, like making the features OS agnostic!

### CODE REVIEW

File: jarviscli/plugins/market_buy.py [hunk 1]
```diff
@@ -68,6 +49,15 @@ def __call__(self, jarvis, s):
...
                 for f in files:
                     if f == 'requirements.txt':
-                        os.system('env/bin/pip install -U -r ' + os.path.join(root, f))
+                        jarvis_folder = os.path.split(self.MARKETPLACE_PATH)[0]
+                        os.system((f'cd {jarvis_folder} && ' +
+                                   'env/bin/pip install -U -r ' +
+                                   os.path.join(root, f)))
+
+        # Deleting the .git folder from the cloned repo.
+        shutil.rmtree(os.path.join(plugin_dir, '.git'))
```
Comment (pnhofmann) [inline jarviscli/plugins/market_buy.py:58]: Why? Don't see benefit - doesn't this just make updates complicated?

Apart from that, good work! Really good idea to use python-git binding instead os.system ;).

File: jarviscli/plugins/market_buy.py [hunk 2]
```diff
@@ -68,6 +49,15 @@ def __call__(self, jarvis, s):
...
+
+        if self.MAINTAIN_PYTHON_ONLY:
```
Comment (pnhofmann) [inline jarviscli/plugins/market_buy.py:60]: Actually, don't see a benefit; something files like LICENSE or README don't waste any significant space and are nothing I would delete.

**Comment by Martins6** [comment]
After a long time...I finished the refact of the code! And the projects keeps growing and that is so cool to see! 

Also, @pnhofmann, I've also created a more simple template for the marketplace in: https://github.com/Martins6/jarvis_marketplace_template

**Comment by pnhofmann** [comment]
Thank you!

> And the projects keeps growing and that is so cool to see!

Yeah, I wanted to merge dev branch by the end of last year; now it's in the middle of march ;). But it'll happen eventually...

**Pull request merged**